### PR TITLE
Blog app - Add forms

### PR DESCRIPTION
**Pull request by Hamzaoutdoors**
# In this milestone we worked on the following 💪🏼  : 

- [x]  added Forms 

 > Additional description of the project and its features.

 - Created forms  to create new `Post`, `Comment` and `Like`
 - Added Create action in every controller to help us create new `Post`, `Comment` and `Like`
 - Rendered views 

- [x]  Added Linters for the project and fix offenses.

##  Build with : 

  - Ruby on Rails 💎 🛤️ 
  - Terminal & RSpec
  - Postgres

------

**N.B**: I used Bootstrap 5 in my app so if you want to see my style you need to run `npm install` first

_THANKS FOR YOUR FEEDBACK 💟_

### CODE REVIEW

File: app/controllers/comments_controller.rb [hunk 1]
```diff
@@ -1,6 +1,13 @@
...
+    @post = Post.find_by_id(params[:post_id])
+    @post.comments.create(comment_params.merge(author_id: current_user.id))
+    redirect_back(fallback_location: root_path)
+  end
+
+  private
+
+  def comment_params
+    params.require(:data).permit(:text)
   end
```
Comment (Dipeshtwis) [inline app/controllers/comments_controller.rb:16]: - [x] The comments controller should have new actions similar to the one shown below. The `new` is used with the HTTP verb GET and creates an object instance.

```ruby

def new
    @comment = Comment.new
  end
```

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -1,25 +1,30 @@
...
+    @post.author_id = current_user.id if current_user
+    @post.title = params[:title]
+    @post.text = params[:text]
+    if @post.save
+      flash[:success] = 'Article Successfully Created'
+      redirect_back(fallback_location: root_path)
+    else
+      render 'new'
+      flash[:info] = 'Create new post'
+    end
```
Comment (Dipeshtwis) [inline app/controllers/posts_controller.rb]: - [x] Please create private methods you can call it `post_params` and use it anywhere you want to use `title` and `text`. Just like you did in `comment` controller.

```ruby
  def post_params
    params.require(:post).permit(:title, :text)
  end
```

**Review by Dipeshtwis — CHANGES_REQUESTED** [review]
Hi @Hamzaoutdoors, 

:+1: This is good to make it this far, your project is really nice, congratulations on the good work you have done so far.

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


 ## Required Changes ♻️

**_Check the comments under the review_**

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- N.A.

Cheers and Happy coding!👏👏👏
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**_Dipesh Kumar_**


------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project(check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/comments_controller.rb (hunk 1 above)
Comment (Hamzaoutdoors) [reply app/controllers/comments_controller.rb:16]: Done ☑️ .. Thanks for feedbacks

File: app/controllers/posts_controller.rb (hunk 2 above)
Comment (Hamzaoutdoors) [reply app/controllers/posts_controller.rb]: Done ☑️  .. Thanks for feedback

Comment (DeliceLydia) [reply app/controllers/posts_controller.rb]: well done

File: app/controllers/comments_controller.rb (hunk 1 above)
Comment (DeliceLydia) [reply app/controllers/comments_controller.rb:16]: well addressed this

**Review by DeliceLydia — APPROVED** [review]
# Congratulations🎉🎉

## Hi @Hamzaoutdoors, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: my highlights for your project:
- PR is professional
- Project is working as expected
- excellent communication with reviewers
Congratulations! 🎉 



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard).. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**